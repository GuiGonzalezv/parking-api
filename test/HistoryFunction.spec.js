const mongoose = require('mongoose')
const {fnParking} = require('../src/functions')
const {BadRequest, NotFound} = require("http-errors")
const db = require('../src/Database/connection')

describe("Get history of reservation by plate", () => {
    it("Returns bad request if doesn't receive a license plate", async () => {
        await expect(fnParking.history()).rejects.toEqual(new BadRequest("Plate is required"))
    })

    it("Returns bad request if license plate is invalid", async () => {
        await expect(fnParking.history("AAAA-333")).rejects.toEqual(new BadRequest("Invalid Plate"))
    })

    it("Returns bad request if no license plate is founded", async () => {
        await expect(fnParking.history("AAA-3333")).rejects.toEqual(new NotFound("Plate not found in our database"))
    })

    it("Return a array of objects with the history of the license plate on parking lot", async() => {
        
        const response = await fnParking.park("UUU-2222")
        await fnParking.payment(response.reservation)
        await fnParking.out(response.reservation)
        const history = await fnParking.history("UUU-2222")

        expect(history[0]).toMatchSnapshot({
            _id: expect.any(mongoose.Types.ObjectId),
            paid: expect.any(Boolean),
            left: expect.any(Boolean),
            time: expect.any(String)
        })
        db.connection.close()
    })
})