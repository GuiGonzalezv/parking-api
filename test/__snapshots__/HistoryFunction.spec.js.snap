// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get history of reservation by plate Return a array of objects with the history of the license plate on parking lot 1`] = `
Object {
  "_id": Any<ObjectId>,
  "left": Any<Boolean>,
  "paid": Any<Boolean>,
  "time": Any<String>,
}
`;
